### PR TITLE
Add student to course

### PR DESCRIPTION
**Pull request by lincAlinc**

### CODE REVIEW

File: app/models/users_course.rb [hunk 1]
```diff
@@ -14,4 +14,6 @@
 class UsersCourse < ApplicationRecord
   belongs_to :user
   belongs_to :course
+
+  enum role: %i[curator mentor lecturer student]
```
Comment (Gklmvch) [inline app/models/users_course.rb]: May be you should extract role array to constant and add collection validation

File: app/models/users_course.rb [hunk 2]
```diff
@@ -14,4 +14,6 @@
 class UsersCourse < ApplicationRecord
```
Comment (Gklmvch) [inline app/models/users_course.rb]:
```suggestion
# frozen_string_literal: true

class UsersCourse < ApplicationRecord
  ACCEPTABLE_ROLES = %i[curator mentor lecturer student].freeze
```

File: app/models/users_course.rb (hunk 1 above)
Comment (Gklmvch) [inline app/models/users_course.rb]:
```suggestion
  enum role: ACCEPTABLE_ROLES
  
  validates_inclusion_of :role, in: ACCEPTABLE_ROLES
```

File: app/controllers/users_courses_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+class UsersCoursesController < ApplicationController
+  before_action :set_course
+  def index
+    @users_courses = UsersCourse.where(course_id: @course)
+    @users = @users_courses.map { |u| User.find(u.user_id) }
+    @users_course = UsersCourse.new
```
Comment (BigiDigi89) [inline app/controllers/users_courses_controller.rb]: @users = @course.users ??
optional students, pluck(:id) or @course.user_ids?

**Pull request merged**